modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -151,33 +151,39 @@ def check_kill_process():
 def initialize():
     """
     Initialize the storescp and Niffler AET.
-    """
-    global niffler_processes
-    global storescp_processes
-    for _ in os.popen("ps ax | grep " + niffler_str + " | grep -v grep"):
-        niffler_processes += 1
-    for _ in os.popen("ps ax | grep -E " + nifflerscp_str + " | grep -v grep"):
-        storescp_processes += 1
-
-    logging.info("Number of running Niffler processes: {0} and storescp processes: {1}".format(niffler_processes,
+    """                                                                          
+    query_aet = QUERY_AET.split(":")[0]
+    dest_aet = DEST_AET
+    if query_aet == dest_aet:
+        global niffler_processes
+        global storescp_processes
+        for _ in os.popen("ps ax | grep " + niffler_str + " | grep -v grep"):
+            niffler_processes += 1
+        for _ in os.popen("ps ax | grep -E " + nifflerscp_str + " | grep -v grep"):
+            storescp_processes += 1
+
+        logging.info("Number of running Niffler processes: {0} and storescp processes: {1}".format(niffler_processes,
                                                                                                storescp_processes))
 
-    if niffler_processes > MAX_PROCESSES:
-        logging.error("[EXTRACTION FAILURE] {0}: Previous {1} extractions still running. As such, your extraction "
+        logging.info("{0}: StoreScp process for the current Niffler extraction is starting now".format(
```

Comment:
Hi,
Thanks for your second attempt.

What I am having a hard time figuring out is why did you move this log up?

With this change, now this log will be printed even before the "if niffler_processes > MAX_PROCESSES:" and "if storescp_processes >= niffler_processes:" processes are checked.

Therefore, the log becomes useless (and always printed), whereas previously, it was printed only when it was actually starting the storescp (previous lines 175 - 176).

My suggestions here (and in your previous PR that I closed) are along two lines:
1. Avoid changing things that already work. That should be a separate PR if you have a reason to do so. A PR should be minimal, addressing the problem at hand only.
2. Make sure you don't introduce new bugs while fixing a previous PR.